extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -1035,34 +1037,33 @@ private void possiblyResetOffsetsOrWait(
     }
   }
 
-  private void resetOffset(
-      KafkaConsumer<byte[], byte[]> consumer,
-      Set<Integer> assignment,
-      TopicPartition topicPartition
-  )
+  private void sendResetRequest(Map<TopicPartition, Long> outOfRangePartitions, TaskToolbox taskToolbox) throws IOException
   {
-    log.warn(
-        "Resetting consumer offset to [%s] for partition [%d]",
-        ioConfig.isUseEarliestOffset() ? "earliest" : "latest",
-        topicPartition.partition()
-    );
-    if (ioConfig.isUseEarliestOffset()) {
-      consumer.seekToBeginning(topicPartition);
-    } else {
-      consumer.seekToEnd(topicPartition);
+    Map<Integer, Long> partitionOffsetMap = Maps.newHashMap();
+    for (Map.Entry<TopicPartition, Long>  outOfRangePartition: outOfRangePartitions.entrySet()) {
+      partitionOffsetMap.put(outOfRangePartition.getKey().partition(), outOfRangePartition.getValue());
     }
-    nextOffsets.put(topicPartition.partition(), consumer.position(topicPartition));
-    log.warn("Consumer is now at offset [%d]", nextOffsets.get(topicPartition.partition()));
-    // check if we seeked passed the endOffset for this partition
-    if (nextOffsets.get(topicPartition.partition()) >= endOffsets.get(topicPartition.partition())
-        && assignment.remove(topicPartition.partition())) {
-      log.info(
-          "Finished reading topic[%s], partition[%,d].",
-          topicPartition.topic(),
-          topicPartition.partition()
-      );
+    boolean result = taskToolbox.getTaskActionClient()
+                                .submit(new ResetDataSourceMetadataAction(
+                                    getDataSource(),
+                                    new KafkaDataSourceMetadata(new KafkaPartitions(
+                                        ioConfig.getStartPartitions()
+                                                .getTopic(),
+                                        partitionOffsetMap
+                                    ))
+                                ));
+
+    if (result) {
+      log.warn("Successfully sent the reset request for partitions [%s], waiting to be killed", partitionOffsetMap.keySet());
+      // wait for being killed by supervisor
+      try {
+        Thread.sleep(Long.MAX_VALUE);
+      }
+      catch (InterruptedException e) {
+        throw new RuntimeException("Got interrupted while waiting to be killed");
+      }
+    } else {
+      log.makeAlert("Failed to send reset request for partitions [%s]", partitionOffsetMap.keySet()).emit();
```

Comment:
what happens to the task after this? ... it can't make progress... shouldn't this throw exception ?